lib/ansible/modules/network/onyx/onyx_ntp.py

```diff
@@ -0,0 +1,240 @@
+    type: list
+    description:
+      - List of ntp authentication keys
+    suboptions:
+      auth_key_id:
+        description:
+          - Configures ntp key-id, range 1-65534
+        required: true
+        type: int
+      auth_key_encrypt_type:
+        description:
+          - encryption type used to configure ntp authentication key.
+        required: true
+        choices: ['md5', 'sha1']
+        type: str
+      auth_key_password:
+        description:
+          - password used for ntp authentication key.
+        required: true
+        type: str
+      auth_key_delete:
+         description:
+             - Used to decide if you want to delete given ntp key or not
+         default: 'no'
+         choices: ['yes', 'no']
+         type: str
+  trusted_keys:
+    type: list
+    description:
+      - List of ntp trusted keys
+"""
+
+EXAMPLES = """
+- name: configure NTP
+  onyx_ntp:
+    state: enabled
+    authenticate_state: enabled
+    ntp_authentication_keys:
+            - auth_key_id: 1
+              auth_key_encrypt_type: md5
+              auth_key_password: 12345
+              auth_key_delete: no
+    trusted_keys: 1,2,3
+"""
+
+RETURN = """
+commands:
+  description: The list of configuration mode commands to send to the device
+  returned: always.
+  type: list
+  sample:
+    - ntp enable
+    - ntp disable
+    - ntp authenticate
+    - no ntp authenticatee
+    - ntp authentication-key 1 md5 12345
+    - no ntp authentication-key 1
+    - ntp trusted-key 1,2,3
+"""
+
+
+from ansible.module_utils.basic import AnsibleModule
+
+from ansible.module_utils.network.onyx.onyx import BaseOnyxModule
+from ansible.module_utils.network.onyx.onyx import show_cmd
+
+
+class OnyxNTPModule(BaseOnyxModule):
+
+    def init_module(self):
+        """ module initialization
+        """
+        ntp_authentication_key_spec = dict(auth_key_id=dict(type='int', required=True),
+                                           auth_key_encrypt_type=dict(required=True, choices=['md5', 'sha1']),
+                                           auth_key_password=dict(required=True),
+                                           auth_key_delete=dict(default='no', choices=['yes', 'no']))
+        element_spec = dict(
+            state=dict(choices=['enabled', 'disabled']),
+            authenticate_state=dict(choices=['enabled', 'disabled']),
+            ntp_authentication_keys=dict(type='list', elements='dict', options=ntp_authentication_key_spec),
+            trusted_keys=dict(type='list', elements='int')
+        )
+        argument_spec = dict()
+        argument_spec.update(element_spec)
+        self._module = AnsibleModule(
+            argument_spec=argument_spec,
+            supports_check_mode=True)
+
+    def _validate_key_id(self):
+        if self._required_config.get("ntp_authentication_keys"):
+            keys_id_list = self._required_config.get("ntp_authentication_keys")
+            for key_item in keys_id_list:
```

Comment:
you should count keys, and restrict number of keys that exceeds limit

Reply:
> you should count keys, and restrict number of keys that exceeds limit

sorry I think the err message is not correct, I meant to write a message that indicates that the key is not valid, there is no max limit needed here